src/shared/lib/GenericAssayUtils/GenericAssayCommonUtils.ts

```diff
@@ -216,42 +249,120 @@ export function makeGenericAssayOption(
     } else {
         label = `${name} (${meta.stableId}): ${description}`;
     }
+    return label;
+}
 
-    if (isPlotsTabOption) {
-        return {
-            value: meta.stableId,
-            label: label,
-            plotAxisLabel: getGenericAssayMetaPropertyOrDefault(
-                meta,
-                COMMON_GENERIC_ASSAY_PROPERTY.NAME,
-                meta.stableId
-            ),
-        };
+export function formatGenericAssayCompactLabelByNameAndId(
+    stableId: string,
+    name: string
+) {
+    const uniqueName = name !== stableId;
+    let label = stableId;
+    if (uniqueName) {
+        label = `${name} (${stableId})`;
     }
-    return {
-        value: meta.stableId,
-        label: label,
-    };
+    return label;
 }
 
-export function getGenericAssayMetaPropertyOrDefault(
-    meta: GenericAssayMeta,
+export function getGenericAssayPropertyOrDefault(
+    genericAssayProperties: {},
     property: string,
     defaultValue: string = NOT_APPLICABLE_VALUE
 ): string {
-    if (property in meta.genericEntityMetaProperties) {
-        return (meta.genericEntityMetaProperties as {
+    if (property in genericAssayProperties) {
+        return (genericAssayProperties as {
             [property: string]: string;
         })[property];
     } else {
         return defaultValue;
     }
 }
 
+export function getGenericAssayMetaPropertyOrDefault(
+    meta: GenericAssayMeta,
+    property: string,
+    defaultValue: string = NOT_APPLICABLE_VALUE
+): string {
+    return getGenericAssayPropertyOrDefault(
+        meta.genericEntityMetaProperties,
+        property,
+        defaultValue
+    );
+}
+
 export function getCategoryOrderByGenericAssayType(genericAssayType: string) {
     // return category order for reserved generic assay type
     // return undefined for other types
     return genericAssayType in RESERVED_CATEGORY_ORDER_DICT
         ? RESERVED_CATEGORY_ORDER_DICT[genericAssayType]
         : undefined;
 }
+
+export function filterGenericAssayEntitiesByGenes(
+    genericAssayEntities: GenericAssayMeta[],
+    hugoGeneSymbols: string[]
+) {
+    // filter logic is: stableId, name or description
+    // is any of them includes sepcific gene, then it's related, we should keep them and filter out others
+    return _.filter(genericAssayEntities, meta =>
+        _.some(
+            hugoGeneSymbols,
+            hugoGeneSymbol =>
+                meta.stableId
```

Comment:
in general i think it's better to use case insensitive regular expressions than do all this lowercasing